test/invariants/fuzz/BreadfundFuzz_DepositWithdraw.t.sol

```diff
@@ -0,0 +1,285 @@
+ * to stress accounting invariants.
+ *
+ * Properties checked:
+ *  - Deposits only once per epoch per member (`AlreadyDeposited`).
+ *  - Withdrawals respect daily contribution, thresholds, and per-epoch limits.
+ *  - Large withdrawals generate requests (and execute only after contest window).
+ *  - Soft conservation invariant (`contract balance + executed ≥ withdrawables`)
+ *    holds at each step.
+ *  - Epoch warp advances counters/reset correctly.
+ *
+ * Notes for reviewers:
+ *  - Some checks are unit-like but are *critical safety guards* (e.g.,
+ *    “already deposited this epoch”).
+ *  - After fixing BUG #2 (large exec not decrementing withdrawables),
+ *    conservation checks here should be switched to *strict* invariant.
+ * ────────────────────────────────────────────────────────────────────────────────
+ */
+
+import {BreadfundFuzzBase} from "./BreadfundFuzzBase.t.sol";
+import {IBreadfund} from "src/interfaces/IBreadfund.sol";
+
+contract BreadfundFuzz_DepositWithdraw is BreadfundFuzzBase {
+  /// -------------------------------------------------------------------------
+  /// Scenario: Soak test over multiple epochs and many randomized operations.
+  /// Properties:
+  ///  - Deposits: only one per member per epoch; sets epoch flag.
+  ///  - Small withdrawals: only if within limit, amount ≤ autoThreshold,
+  ///    and ≤ withdrawable balance.
+  ///  - Large withdrawals: create requests, not executed until contest window.
+  ///  - Conservation (soft) holds across ops and epochs.
+  /// -------------------------------------------------------------------------
+  function testFuzz_Soak_ManyDepositsAndWithdrawals(
+    uint8 epochsRaw, uint8 opsRaw, uint256 extraSeed
+  ) public {
+    uint256 epochs = bound(uint256(epochsRaw), 2, 12);
+    uint256 ops    = bound(uint256(opsRaw),    5, 40);
+
+    IBreadfund.Breadfund memory cfg = safeCfg;
+    cfg.breadfundStart = block.timestamp;
+    cfg.members        = defaultMembers;
+    cfg.ratio          = 1;
+    uint256 id = breadfund.create(cfg);
+
+    _mintApprove(member1, 1e24, address(breadfund));
+    _mintApprove(member2, 1e24, address(breadfund));
+    _mintApprove(member3, 1e24, address(breadfund));
+
+    uint256 seed = uint256(keccak256(abi.encodePacked(block.timestamp, epochs, ops, extraSeed)));
+    _assertConservative(id, defaultMembers);
+
+    for (uint256 e = 0; e < epochs; e++) {
+      for (uint256 k = 0; k < ops; k++) {
+        seed = uint256(keccak256(abi.encodePacked(seed, e, k)));
+        address actor = _pick(defaultMembers, seed);
+        uint256 roll  = seed % 10;
+
+        uint256 epochIdx   = breadfund.getCurrentEpochIndex(id);
+        uint256 contrib    = breadfund.breadfundMemberContribute(id, actor);
+        uint256 withdrawableBefore = breadfund.memberWithdrawableBalance(id, actor);
+
+        // ── Case 1: Deposits ───────────────────────────────────────────────
+        if (roll <= 2) {
```

Comment:
personally don't like this "roll" approach. would rather just see the different cases have their own tests